src/charm.py

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env python3
+# Copyright 2023 Mariyan Dimitrov
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Charm the service.
+
+Refer to the following post for a quick-start guide that will help you
+develop a new k8s charm using the Operator Framework:
+
+https://discourse.charmhub.io/t/4208
+"""
+
+import collections
+import logging
+import os
+
+import ops.lib
+from charms.nginx_ingress_integrator.v0.nginx_route import require_nginx_route
+from ops.charm import CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+pgsql = ops.lib.use("pgsql", 1, "postgresql-charmers@lists.launchpad.net")
+DATABASE_NAME = "synapse"
+
+VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+
+
+class MatrixOperatorCharm(CharmBase):
+    """Charm the service."""
+
+    _CONTAINER_NAME = "synapse"
+    _SYNAPSE_CONFIG_PATH = "/data/homeserver.yaml"
+
+    _stored = StoredState()
```

Comment:
Is this necessary? The recommendation is currently against using this and instead using a file on the application container

Reply:
It's reusing code from our charms, I'm fine to change it. Can you post a link to the recommendation?